docs/providers/spark.md

```diff
@@ -0,0 +1,76 @@
+# Spark 
+
+Featureform supports [Spark](https://spark.apache.org/) as an Offline Store.
+
+This means that Featureform can handle all flavors of Spark using S3, GCS, Azure Blob Store or HDFS as a backing store.
+
+Common use cases of Featureform with Spark include: 
+
+{% content-ref url="../providers/spark_emr.md" %}
+[spark_emr.md](../providers/spark_emr.md)
+{% endcontent-ref %}
+
+{% content-ref url="../providers/spark_databricks.md" %}
+[spark_databricks.md](../providers/spark_databricks.md)
+{% endcontent-ref %}
+
+## Understanding The Different Flavors of Spark <a href="#implementation" id="implementation"></a>
+
+Spark is a powerful, open-source general computing framework developed for large-scale data processing.
+
+Databricks is a managed data and analytics platform developed on top of Spark.
+
+Both Spark and Databricks can be self-hosted in Kubernetes and other non-cloud implementations, as well as hosted with popular cloud providers such as AWS, Azure, GCP and on Databricks.
+
+| Catgeory  | Featureform Supported  |
+|---|---|
+| Types of Spark  | Spark, Databricks  |
+| Specific Cloud Offerings  | EMR |
+| Filestores  | S3, GCS, Azure Blobs, HDFS  |
+
+
+## Transforming Data
+
+### Transformation Sources
+
+Using Spark and a filestore (GCS, Azure Blob Storage, S3, HDFS) as an Offline Store, you can [define new transformations](../getting-started/transforming-data.md) via [SQL and Spark DataFrames](https://spark.apache.org/docs/latest/sql-programming-guide.html). Using either these transformations or pre-existing files in your file store, a user can chain transformations and register columns in the resulting tables as new features and labels.
```

Comment:
for the file stores, I wonder if it makes sense to link to their sdk docs once we have them? also, we have filestore and file store used throughout the docs.

Reply:
100%, we should link the SDK docs. 

Will settle on `filestores`.

Reply:
For the links to the SDK docs, going to merge what we have right now so I'm not blocking anyone